test/hotspot/jtreg/compiler/c2/irTests/IntegerDivValueTests.java

```diff
@@ -0,0 +1,302 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package compiler.c2.gvn;
+
+import compiler.lib.generators.*;
+import compiler.lib.ir_framework.*;
+import jdk.test.lib.Asserts;
+
+/*
+ * @test
+ * @bug 8364766
+ * @summary Test value method of DivINode and DivLNode
+ * @key randomness
+ * @library /test/lib /
+ * @run driver compiler.c2.gvn.IntegerDivValueTests
+ */
+public class IntegerDivValueTests {
+    private static final RestrictableGenerator<Integer> INTS = Generators.G.ints();
+    private static final RestrictableGenerator<Long> LONGS = Generators.G.longs();
+
+    public static void main(String[] args) {
+        TestFramework.run();
+    }
+
+    @Test
+    @IR(failOn = {IRNode.DIV_I, IRNode.DIV_L})
+    public int testIntConstantFolding() {
+        // All constants available during parsing
+        return 50 / 25;
```

Comment:
This will be constant-folded by javac, so won't exercise c2

Reply:
Whoops, thanks! Fixed in the latest commit